From 1b77a4c27b8a3297a645fcb0b92ca76d405beacb Mon Sep 17 00:00:00 2001
From: Thomas Monjalon <thomas@monjalon.net>
Date: Sun, 14 Feb 2010 20:41:35 +0100
Subject: [PATCH] rename option -w to -t (for timeout)

Now timeout is disabled by default.
It must specified if really needed (e.g. in scripts).
---
 README |  4 ++--
 clunc  | 15 +++++++--------
 2 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/README b/README
index 0081381..8d2aa15 100644
--- a/README
+++ b/README
@@ -56,8 +56,8 @@ It can be used
      as an interactive shell
   or to send a batch of commands via the standard input
 
-In case of batch commands, CLUNC will close itself after a inactivity time.
-This timeout can be defined with option "-w". The default is 1 second.
+In case of batch commands, CLUNC can close itself after a inactivity time.
+This timeout can be defined with option "-t".
 
 Examples:
     clunc -i 192.168.42.3
diff --git a/clunc b/clunc
index e28d86a..9dba8ea 100755
--- a/clunc
+++ b/clunc
@@ -25,8 +25,7 @@ print_usage () {
 		         -v           verbose
 		         -m address   MAC to target
 		         -i address   IP to set (mandatory)
-		         -w seconds   inactivity time to wait
-		                      before closing a batch session (default: 1)
+		         -t seconds   close batch session after an inactivity time
 	END_OF_USAGE
 }
 
@@ -42,7 +41,7 @@ set_defaults () {
 	unset IP
 	SERVER_PORT=6666
 	CLIENT_PORT=6666
-	DELAY=1
+	unset TIMEOUT_OPT
 	VERBOSE=0
 }
 
@@ -50,7 +49,7 @@ handle_args () {
 	SELF=$(basename $0)
 	SELF_DIR=$(readlink -em $(dirname $0))
 
-	while getopts 'hvm:i:w:' ARG ; do
+	while getopts 'hvm:i:t:' ARG ; do
 		case $ARG in
 			h )
 				print_help
@@ -67,12 +66,12 @@ handle_args () {
 				IP=$OPTARG
 				OPEN_OPTS="$OPEN_OPTS -i $IP"
 			;;
-			w )
+			t )
 				if ! has_batch_input ; then
-					echo 'option -w is for batch mode'
+					echo 'option -t is for batch mode'
 					syntax_error
 				fi
-				DELAY=$OPTARG
+				TIMEOUT_OPT="-w $OPTARG"
 			;;
 			* )
 				syntax_error
@@ -160,7 +159,7 @@ fi
 print_header
 if has_batch_input ; then # input is a pipe or a file
 	printf 'Marvell>> '
-	$NC_CMD -w $DELAY
+	$NC_CMD $TIMEOUT_OPT
 	echo ; echo
 else # input from a terminal
 	(echo ; cat) | $NC_CMD
-- 
2.14.3 (Apple Git-98)

